packages/components/src/navigation/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Animate from '../animate';
+import { NavigationContext } from './context';
+import { DEFAULT_LEVEL } from './constants';
+import { Root } from './styles/navigation-styles';
+
+export default function Navigation( {
+	activeItem,
+	activeLevel = DEFAULT_LEVEL,
+	children,
+	className,
+	setActiveItem = noop,
+	setActiveLevel = noop,
+} ) {
+	const [ item, setItem ] = useState( activeItem );
+	const [ level, setLevel ] = useState( activeLevel );
+	const [ slideOrigin, setSlideOrigin ] = useState();
+
+	const activateItem = ( itemId ) => {
+		setItem( itemId );
+		setActiveItem( itemId );
```

Comment:
This probably shouldn't fire if `itemId` is non-existent but we currently don't have a way to check that? (same for level below)

Reply:
This makes total sense, but I'd delay to a follow up.

It would require the same children-traversing logic that we'd need to automatize the back button: doable but possibly long.

Reply:
Yup, agreed that we can iterate on this later.